File: utils/request.js
const axios = require('axios')

const http = axios.create({
  timeout: 10000,
  baseURL: 'http://localhost:8010'
})

http.interceptors.response.use((response) => {
  const { code, message } = response.data
  return response
}, err => {
  return Promise.reject(err)
})
const httpPost = http.post
const httpGet = http.get
async function post(url, params = {}, config = {}) {
  const validCode = params.validCode
  if (validCode) {
    delete params.validCode
  }

  const response = await httpPost(url, params, config)
  return response.data
}

function upload(url, params = {}, config = {}) {
  return httpPost(url, {
    params
  }, {
    ...config,
    ...{
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    }
  })
}

async function get(url, params = {}, config = {}) {
  const response = await httpGet(url, { params, ...config })
  return response?.data
}

module.exports = {
  post,
  get,
  upload,
}

File: utils/request.py
import requests

# 创建会话对象
session = requests.Session()
session.timeout = 10
session.base_url = 'http://localhost:8010'

async def post(url, params=None, config=None):
    if params is None:
        params = {}
    if config is None:
        config = {}
    
    # 处理validCode参数
    valid_code = params.pop('validCode', None)
    
    # 构建请求头
    headers = config.get('headers', {})
    headers.setdefault('Content-Type', 'application/json')
    
    try:
        response = session.post(url, json=params, headers=headers)
        response.raise_for_status()
        return response.json()
    except Exception as e:
        raise e

async def get(url, params=None, config=None):
    if params is None:
        params = {}
    if config is None:
        config = {}
    
    try:
        response = session.get(url, params=params, **config)
        response.raise_for_status()
        return response.json()
    except Exception as e:
        raise e

async def upload(url, params=None, config=None):
    if params is None:
        params = {}
    if config is None:
        config = {}
    
    headers = config.get('headers', {})
    headers.setdefault('Content-Type', 'multipart/form-data')
    
    try:
        response = session.post(url, data={'params': params}, headers=headers)
        response.raise_for_status()
        return response.json()
    except Exception as e:
        raise e
